pkg/generator/ingress_translator.go

```diff
@@ -169,15 +180,31 @@ func (translator *IngressTranslator) translateIngress(ctx context.Context, ingre
 				}
 
 				connectTimeout := 5 * time.Second
-				cluster := envoy.NewCluster(splitName, connectTimeout, publicLbEndpoints, http2, typ)
+
+				var transportSocket *envoycorev3.TransportSocket
+
+				// This has to be "OrDefaults" because this path is called before the informers are
+				// running when booting the controller up and prefilling the config before making it
+				// ready.
+				cfg := config.FromContextOrDefaults(ctx)
+
+				// TODO: As Ingress with RewriteHost points to ExternalService(kourier-internal), we don't need to enable TLS?
+				if activatorCA := cfg.Network.ActivatorCA; activatorCA != "" && httpPath.RewriteHost == "" {
+					var err error
+					transportSocket, err = translator.createUpstreamTransportSocket(activatorCA, config.FromContext(ctx).Network.ActivatorSAN, http2)
+					if err != nil {
+						return nil, err
+					}
+				}
+				cluster := envoy.NewCluster(splitName, connectTimeout, publicLbEndpoints, http2, transportSocket, typ)
 				clusters = append(clusters, cluster)
 
 				weightedCluster := envoy.NewWeightedCluster(splitName, uint32(split.Percent), split.AppendHeaders)
 				wrs = append(wrs, weightedCluster)
 			}
 
 			if len(wrs) != 0 {
-				// disable ext_authz filter for HTTP01 challenge when the feature is enabled
+				// disable ext_tlsz filter for HTTP01 challenge when the feature is enabled
```

Comment:
This is confusing to me. My understanding is that what we disable is achieved when passing `envoy.NewRouteExtAuthzDisabled`, so not sure why change to `ext_tlsz`.
In addition it is a bit weird to have a condition such as `if extAuthzEnabled` and when that is satisfied to pass `envoy.NewRouteExtAuthzDisabled`. The names used are contradictory.

Reply:
This comment was wrongly changes when I updated the lib alias name by `s/auth/tls/g`.
So, no change around this code. I think it is not wrong for the current code but please send the PR if you found something wrong.

Reply:
ok thanks